source/text_handler.cpp

```diff
@@ -147,9 +141,8 @@ spv_result_t getWord(spv_text text, spv_position position, std::string& word,
           if (escaping || quoting) break;
         // Fall through.
         case '\0': {  // NOTE: End of word found!
-          word.assign(
-              text->str + position->index,
-              static_cast<size_t>(endPosition->index - position->index));
+          word->assign(text->str + start_index,
+                       static_cast<size_t>(position->index - start_index));
```

Comment:
Optional: `assign(text->str + start_index, text->str + position->index)` is shorter and more readable.